diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter5/ParameterNameInOuterLambdaRule.kt

```diff
@@ -24,9 +29,19 @@ class ParameterNameInOuterLambdaRule(configRules: List<RulesConfig>) : DiktatRul
     }
 
     private fun checkLambda(node: ASTNode) {
-        val hasInnerLambda = node
-            .findAllDescendantsWithSpecificType(KtNodeTypes.LAMBDA_EXPRESSION, false)
-            .isNotEmpty()
+        val configuration = ParameterNameInOuterLambdaConfiguration(
+            configRules.getRuleConfig(PARAMETER_NAME_IN_OUTER_LAMBDA)?.configuration
+                ?: emptyMap()
+        )
+        val strictMode = configuration.strictMode
+
+        val innerLambdaList = node.findAllDescendantsWithSpecificType(KtNodeTypes.LAMBDA_EXPRESSION, false)
+        val hasInnerLambda = innerLambdaList.isNotEmpty()
+
+        if (!strictMode && (hasNoParameters(node) || node.hasExplicitIt()) && (!hasInnerLambda ||
+                innerLambdaList.all { innerLambda -> !hasItInLambda(innerLambda) })) {
```

Comment:
Maybe it would be better to extract this condition into variable or method and give it clearer name?

Reply:
Done